### ISSUE TITLE
Cannot run game while using scrcpy, security violation.

### ISSUE THREAD

**Issue by plastikhub** [Genymobile/scrcpy#2899]
- [x] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
- [x] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: macOS
 - scrcpy version: 1.18
 - installation method: brew
 - device model: Huawei
 - Android version: 5.0.1

**Describe the bug**
scrcpy is working fine on the device until I try to load a game, then I get the following error: Security Violation: 'ADB or ROOT Shells (ex: /System/bin/sh)' is running in the background! This app will be terminated!

After, the game will shut down. I can't seem to figure out how to bypass this. Please help!

**Comment by rom1v** [comment]
The game checks if adb is enabled and decides to terminate itself.

**Comment by plastikhub** [comment]
This only happens on one device. I am also using the same game same build on a Motorola device and it works perfect with scrcpy. Is there any workaround?

**Comment by rom1v** [comment]
On this particular device, if `adb` is enabled, but you don't run scrcpy, does it still terminate?

**Comment by plastikhub** [comment]
Are you referring to debugging inside developer?

**Comment by rom1v** [comment]
Yes.

Basically, if you just don't run scrcpy without changing anything else, does your game terminate?

**Comment by plastikhub** [comment]
No, it only terminates when running scrcpy. Debugging is ON and it works fine.

**Comment by rom1v** [comment]
OK, I don't know on what criteria the game decides to terminate itself.

If you just mirror without injecting input events (`-n`/`--no-control`), does it work?

```
scrcpy -n
```

**Comment by plastikhub** [comment]
Ok success! So, just to try something stupid, pulling down the notifications, I changed the USB MODE to "USB Drive" instead of the default "Files" and that seemed to work. The only problem is it is disconnecting itself from the computer after a certain time-frame. This could be cable issue or something else, so I will test that. But, it looks like some sort of progression at least. It could be that this phone is just old and finicky as well (the disconnect/reconnect).

**Comment by plastikhub** [comment]
Unfortunately, counted my chickens too soon. Not a reliable fix as I am still having the issue in either mode.
